diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4d9dd85..0b49edc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,7 @@ cmake_minimum_required(VERSION 3.16)
 
 project(BoringSSL LANGUAGES C CXX)
 
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_C_STANDARD 11)
 set(CMAKE_C_STANDARD_REQUIRED ON)
diff --git a/src/crypto/spake2plus/spake2plus.cc b/src/crypto/spake2plus/spake2plus.cc
index c191855..ee41b20 100644
--- a/src/crypto/spake2plus/spake2plus.cc
+++ b/src/crypto/spake2plus/spake2plus.cc
@@ -130,8 +130,8 @@ bool ComputeTranscript(uint8_t out_prover_confirm[kConfirmSize],
   ScalarToSizedBuffer(group, w0, w0_enc);
 
   uint8_t K_main[SHA256_DIGEST_LENGTH];
-  UpdateWithLengthPrefix(sha, Span(prover_share, kShareSize));
-  UpdateWithLengthPrefix(sha, Span(verifier_share, kShareSize));
+  UpdateWithLengthPrefix(sha, Span<const uint8_t>(prover_share, kShareSize));
+  UpdateWithLengthPrefix(sha, Span<const uint8_t>(verifier_share, kShareSize));
   UpdateWithLengthPrefix(sha, Z_enc);
   UpdateWithLengthPrefix(sha, V_enc);
   UpdateWithLengthPrefix(sha, w0_enc);
diff --git a/src/include/openssl/span.h b/src/include/openssl/span.h
index fa77962..7013f4c 100644
--- a/src/include/openssl/span.h
+++ b/src/include/openssl/span.h
@@ -71,8 +71,8 @@ class SpanBase {
 // a Span<T> by checking for data() and size() member functions.
 template <typename C, typename T>
 using EnableIfContainer = std::enable_if_t<
-    std::is_convertible_v<decltype(std::declval<C>().data()), T *> &&
-    std::is_integral_v<decltype(std::declval<C>().size())>>;
+    std::is_convertible<decltype(std::declval<C>().data()), T *>::value &&
+    std::is_integral<decltype(std::declval<C>().size())>::value>;
 
 }  // namespace internal
 
@@ -203,6 +203,8 @@ class Span : private internal::SpanBase<const T> {
 template <typename T>
 const size_t Span<T>::npos;
 
+// Deduction guides require C++17
+#if __cplusplus >= 201703L
 template <typename T>
 Span(T *, size_t) -> Span<T>;
 template <typename T, size_t size>
@@ -212,6 +214,7 @@ template <
     typename T = std::remove_pointer_t<decltype(std::declval<C>().data())>,
     typename = internal::EnableIfContainer<C, T>>
 Span(C &) -> Span<T>;
+#endif
 
 template <typename T>
 constexpr Span<T> MakeSpan(T *ptr, size_t size) {
