--- a/src/art/libartbase/base/mem_map.cc
+++ b/src/art/libartbase/base/mem_map.cc
@@ -44,6 +44,20 @@
 #define MAP_ANONYMOUS MAP_ANON
 #endif
 
+// see aosp/build/soong/android/config.go
+/*
+func (c *config) LibartImgHostBaseAddress() string {
+	return "0x60000000"
+}
+
+func (c *config) LibartImgDeviceBaseAddress() string {
+	return "0x70000000"
+}
+*/
+
+// device base address
+#define ART_BASE_ADDRESS 0x70000000
+
 namespace art {
 
 using android::base::StringPrintf;
