--- a/src/openscreen/platform/api/task_runner.h
+++ b/src/openscreen/platform/api/task_runner.h
@@ -24,11 +24,14 @@
  public:
 // Seem to get an error using clang when compiling with -fno-exceptions:
 //   error: implicit instantiation of undefined template 'std::__1::packaged_task<void () noexcept>'
+/*
 #if __has_feature(cxx_exceptions)
   using Task = std::packaged_task<void() noexcept>;
 #else
   using Task = std::packaged_task<void()>;
 #endif
+*/
+  using Task = std::packaged_task<void()>;
 
   virtual ~TaskRunner() = default;
 
